### ISSUE TITLE
mp3 encoder?

### ISSUE THREAD

**Issue by coodoo** [ai/audio-recorder-polyfill#15]
First of all, thanks for building this great polyfill, it's a life saver on safari.

I saw there's a wav encoder built-in, wondering is there any chance for a mp3 encoder, maybe using [lamejs](https://github.com/zhuker/lamejs), so that recorded audio files could be played across chrome/firefox/safari?

**Comment by ai** [comment]
Hm, interesting idea. I personally prefer Ogg Opus, but MP3 could be a easier way.

**Comment by coodoo** [comment]
That was my first thought, but by default Chrome will encode audio with opus codec, but for `lamejs` to work we probably will have to feed `wav` to it?

**Comment by PetterS** [comment]
It is pretty easy to just pass the wave data from this polyfill into lamejs. Replace wave-encoder.js with lamejs. I have been doing that and it works great.

**Comment by ai** [comment]
Can you send PR to docs how to use lamejs?

**Comment by adueck** [comment]
@PetterS Yes, if you could that it would be great! 👏

**Comment by PetterS** [comment]
I did it by parsning the WAV header which is far from the best solution.

I think a much better solution would be to follow `wave-encoder.js` and create a `mp3-encoder.js` as well.

**Comment by ai** [comment]
Released 0.2 version. Thanks to @youkaclub for the work.

**Comment by dmitrybndar** [comment]
Hi guys, thanks for your polyfill and for mp3 encoder, I just wanted to ask is there any workaround to make lamejs options (kbps, bitrate and channel) to be customizable? They are now hardcoded to mono channel, 44100 rate and 128kbps. I think it's nice approach to be able edit them.

**Comment by ai** [comment]
We can use this API:

```js
MediaRecorder.encoder = require('audio-recorder-polyfill/mpeg-encoder')(opts)
```

**Comment by dmitrybndar** [comment]
@ai, cool! Will this work with es6 imports? And when you will can implement this? Thanks.

**Comment by ai** [comment]
@dmitrybndar fixed at 0.3 with [`dual-publish`](https://github.com/ai/dual-publish/)

https://github.com/ai/audio-recorder-polyfill#mp3
